src/components/App/App.js

```diff
@@ -1,26 +1,34 @@
 import './App.scss';
-import { useEffect, useState } from 'react';
-import { getPictureOfDay, getIncomingNearEarthObjects } from '../../apiCalls';
+import { useState } from 'react';
 import IncomingObjects from '../IncomingObjects/IncomingObjects';
 import Nav from '../Nav/Nav';
+import EventDetails from '../EventDetails/EventDetails';
 import ImageOfDay from '../ImageOfDay/ImageOfDay';
 import { Routes, Route } from 'react-router-dom'
 
 function App() { 
-  const [networkError, setNetworkError] = useState(false)
+  const [networkError, setNetworkError] = useState(false);
+  const [eventDetails, setEventDetails] = useState();
 
   const handleNetworkErrors = (error) => {
     setNetworkError(error.message)
   }
 
+  const handleEvent = (data) => {
+
+      setEventDetails(data);
+
+  }
+
   return (
     <div className="app">
       <Nav />
       {!networkError ? 
         <main>
           <Routes>
             <Route path='/' element={<ImageOfDay handleNetworkErrors={handleNetworkErrors}/>} />
-            <Route path='stargazing-events' element={<IncomingObjects handleNetworkErrors={handleNetworkErrors} />} />
+            <Route path='stargazing-events' element={<IncomingObjects handleEvent={handleEvent} handleNetworkErrors={handleNetworkErrors} />} />
+            <Route path='stargazing-events/:jd/:des' element={eventDetails && <EventDetails eventDetails={eventDetails} handleNetworkErrors={handleNetworkErrors} />} />
```

Comment:
To confirm my understanding, you used two dynamic sub url's to use both of them to retrieve some dynamic data in the EventDetails component?

Reply:
No, the second one should not be there... good eye!
I had copied it down to see it the saved events but forgot to change some of the properties.